espnet/nets/beam_search_transducer.py

```diff
@@ -129,71 +161,117 @@ def sort_nbest(
 
         return hyps[: self.nbest]
 
-    def greedy_search(self, h: torch.Tensor) -> List[Hypothesis]:
-        """Greedy search implementation for transformer-transducer.
+    def prefix_search(
+        self, hyps: List[ExtendedHypothesis], enc_out_t: torch.Tensor
+    ) -> List[ExtendedHypothesis]:
+        """Prefix search for NSC and mAES strategies.
+
+        Based on https://arxiv.org/pdf/1211.3711.pdf
+
+        """
+        for j, hyp_j in enumerate(hyps[:-1]):
+            for hyp_i in hyps[(j + 1) :]:
+                curr_id = len(hyp_j.yseq)
+                pref_id = len(hyp_i.yseq)
+
+                if (
+                    is_prefix(hyp_j.yseq, hyp_i.yseq)
+                    and (curr_id - pref_id) <= self.prefix_alpha
+                ):
+                    logp = torch.log_softmax(
+                        self.joint_network(enc_out_t, hyp_i.dec_out[-1])
+                        / self.softmax_temperature,
+                        dim=-1,
+                    )
+
+                    curr_score = hyp_i.score + float(logp[hyp_j.yseq[pref_id]])
+
+                    for k in range(pref_id, (curr_id - 1)):
+                        logp = torch.log_softmax(
```

Comment:
This is on my TODO list, I'll make a PR to reach 100% coverage on Transducer-related files after the next PR (recipes update). If it's okay with you, you can merge current PR, otherwise, I'll add them later in the day.

Reply:
We have several internal implementations, but they are espnet2.
This would be the discussion point.
But anyway, the plan looks good to me!